rules/repositories.bzl

```diff
@@ -116,30 +133,59 @@ def _rules_ios_bzlmod_dependencies():
             repo = "rules_apple",
             sha256 = "1618fc82e556ebc97ea360b8cacd3365ca3b0e0a85ccb32422468204843e752d",
         )
+        _maybe(
+            rules_apple_api,
+            name = "rules_apple_api",
+            version = "1_0",
+        )
+    elif bazel_version.major == "6" and bazel_version.minor == "1":
```

Comment:
Why is this needed? rules_apple 2 and 3 both support Bazel 6.x.

Reply:
We currently test rules_apple 2 with `6.1.2` on CI